src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs

```diff
@@ -73,6 +73,18 @@ public async Task StartAsync(CancellationToken cancellationToken)
                 urls = Options.WebHostOptions.ServerUrls;
             }
 
+            if (string.IsNullOrEmpty(urls))
+            {
+                // HTTP_PORTS and HTTPS_PORTS, these are lower priority than Urls.
+                var httpPorts = Configuration[WebHostDefaults.HttpPortKey];
+                var httpUrls = httpPorts?.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+                    .Select(port => $"http://*:{port}").Aggregate((s1, s2) => string.Concat(s1, ";", s2));
+                var httpsPorts = Configuration[WebHostDefaults.HttpsPortKey];
+                var httpsUrls = httpsPorts?.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+                    .Select(port => $"https://*:{port}").Aggregate((s1, s2) => string.Concat(s1, ";", s2));
+                urls = string.Join(';', httpUrls, httpsUrls);
```

Comment:
```suggestion
                urls = $"{httpUrls};{httpsUrls}";
```

Reply:
A leading or trailing semicolon doesn't matter because the string split below removes empty entries, right?